File: ldfs/src/main/kotlin/com/ldfs/main/ChunkServerController.java
package com.ldfs.main;

import com.ldfs.control.domain.model.entity.ChunkServerEntity;
import com.ldfs.control.domain.service.ChunkServerAccessService;
import com.ldfs.main.dto.request.HeartBeatRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/chunkServer")
public class ChunkServerController {
    private final ChunkServerAccessService chunkServerAccessService;

    @Autowired
    public ChunkServerController(ChunkServerAccessService chunkServerAccessService) {
        this.chunkServerAccessService = chunkServerAccessService;
    }

    @GetMapping
    public ResponseEntity<List<ChunkServerEntity>> getChunkServers() {
        return ResponseEntity.ok().body(chunkServerAccessService.findAll());
    }

    @PostMapping("/heartBeat")
    public ResponseEntity<ChunkServerEntity> heartBeatUpdateServerStatus(@RequestBody HeartBeatRequest heartBeatRequest) {
        ChunkServerEntity storedEntity = chunkServerAccessService.makeServerDiscoverable(heartBeatRequest.ip,
                heartBeatRequest.port,
                heartBeatRequest.remainingStorageSize);
        return ResponseEntity.ok().body(storedEntity);
    }
}

File: ldfs/src/main/kotlin/com/ldfs/main/ChunkServerController.kt
import com.ldfs.control.domain.model.entity.ChunkServerEntity
import com.ldfs.control.domain.service.ChunkServerAccessService
import com.ldfs.main.dto.request.HeartBeatRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/chunkServer")
class ChunkServerController(
    private val chunkServerAccessService: ChunkServerAccessService,
) {
    @GetMapping
    fun chunkServers(): ResponseEntity<List<ChunkServerEntity>> {
        return ResponseEntity.ok().body(chunkServerAccessService.findAll())
    }

    @PostMapping("/heartBeat")
    fun heartBeatUpdateServerStatus(
        @RequestBody heartBeatRequest: HeartBeatRequest,
    ): ResponseEntity<ChunkServerEntity> {
        val storedEntity =
            chunkServerAccessService.makeServerDiscoverable(
                heartBeatRequest.ip!!,
                heartBeatRequest.port!!,
                heartBeatRequest.remainingStorageSize!!,
            )
        return ResponseEntity.ok().body(storedEntity)
    }
}
